src/screens/Activity/TimeRangePrompt.tsx

```diff
@@ -0,0 +1,403 @@
+						})}
+						{' – '}
+						{t('intl:dateTime', {
+							v: range[1],
+							formatParams: {
+								v: { day: 'numeric', month: 'long', year: 'numeric', timeZone },
+							},
+						})}
+					</Text01M>
+				)}
+			</View>
+
+			<View style={styles.buttonContainer}>
+				<Button
+					style={styles.button}
+					size="large"
+					variant="secondary"
+					text={t('filter_clear')}
+					onPress={handleClear}
+					disabled={range.length === 0}
+					testID="CalendarClearButton"
+				/>
+				<View style={styles.divider} />
+				<Button
+					style={styles.button}
+					size="large"
+					text={t('filter_apply')}
+					onPress={handleApply}
+					disabled={range.length === 0}
+					testID="CalendarApplyButton"
+				/>
+			</View>
+		</View>
+	);
+};
+
+const TimeRangePrompt = ({
+	onChange,
+}: {
+	onChange: Function;
+}): ReactElement => {
+	const { t } = useTranslation('wallet');
+	const snapPoints = useSnapPoints('calendar');
+
+	useBottomSheetBackPress('timeRangePrompt');
+
+	const handleClose = (): void => {
+		closeBottomSheet('timeRangePrompt');
+	};
+
+	return (
+		<BottomSheetWrapper
+			view="timeRangePrompt"
+			snapPoints={snapPoints}
+			backdrop={true}
+			onClose={handleClose}>
+			<View style={styles.root}>
+				<Subtitle style={styles.title}>{t('filter_title')}</Subtitle>
+
+				<Calendar onChange={onChange} />
+
+				<SafeAreaInsets type="bottom" />
+			</View>
+		</BottomSheetWrapper>
+	);
+};
+
+const styles = StyleSheet.create({
+	root: {
+		flex: 1,
+	},
+	title: {
+		marginBottom: 25,
+		textAlign: 'center',
+	},
+	calendar: {
+		flex: 1,
+	},
+	selector: {
+		paddingLeft: 16,
+		flexDirection: 'row',
+		alignItems: 'center',
+		justifyContent: 'flex-end',
+	},
+	current: {
+		flexGrow: 1,
+		justifyContent: 'center',
+	},
+	arrow: {
+		width: 45,
+		height: 45,
+		justifyContent: 'center',
+		alignItems: 'center',
+	},
+	weekCaption: {
+		flexDirection: 'row',
+		marginTop: 12,
+	},
+	weekDay: {
+		justifyContent: 'center',
+		alignItems: 'center',
+		flex: 1,
+	},
+	week: {
+		flexDirection: 'row',
+	},
+	day: {
+		height: DAY_HEIGHT,
+		flex: 1,
+		justifyContent: 'center',
+		alignItems: 'center',
+		marginTop: 12,
+	},
+	sRoot: {
+		...StyleSheet.absoluteFillObject,
+		justifyContent: 'center',
+		alignItems: 'center',
+	},
+	sStartBrick: {
+		...StyleSheet.absoluteFillObject,
+		left: '50%',
+	},
+	sEndBrick: {
+		...StyleSheet.absoluteFillObject,
+		right: '50%',
+	},
+	sMiddle: {
+		...StyleSheet.absoluteFillObject,
+	},
+	sCircleMulti: {
+		borderRadius: DAY_HEIGHT,
+		height: DAY_HEIGHT,
+		width: DAY_HEIGHT,
+		backgroundColor: '#502305',
+	},
+	sCircleSingle: {
+		borderRadius: DAY_HEIGHT,
+		height: DAY_HEIGHT,
+		width: DAY_HEIGHT,
+	},
+	buttonContainer: {
+		marginTop: 'auto',
+		flexDirection: 'row',
+		justifyContent: 'center',
+		marginHorizontal: 16,
+	},
+	button: {
+		flex: 1,
+	},
+	divider: {
+		width: 16,
+	},
+	range: {
+		flex: 1,
+		justifyContent: 'center',
+		alignItems: 'center',
+	},
```

Comment:
I would change these styles to avoid the text jumping when the number of rows changes:

```suggestion
	range: {
		marginTop: 'auto',
		marginBottom: 36,
		alignItems: 'center',
	},
```

Reply:
Still jumps for me

Reply:
I forgot, also need to  remove `marginTop: 'auto',` from `buttonContainer` styles

Reply:
Nice, added